lib/SILOptimizer/LoopTransforms/LICM.cpp

```diff
@@ -615,6 +619,13 @@ void LoopTreeOptimization::analyzeCurrentLoop(
         checkSideEffects(Inst, MayWrites);
         break;
       }
+      case SILInstructionKind::RefElementAddrInst: {
+        auto *REA = dyn_cast<RefElementAddrInst>(&Inst);
+        assert(REA && "ref_element_addr");
+        SpecialHoist.insert(REA);
+        checkSideEffects(Inst, MayWrites);
```

Comment:
That's not needed. ref_element_addr cannot have any side effects

Reply:
done